memote/suite/cli/runner.py

```diff
@@ -393,88 +411,282 @@ def history(model, message, rewrite, solver, location, pytest_args, deployment,
+            headers=headers,
+            json=payload
+        )
+        response.raise_for_status()
+    except HTTPError:
         LOGGER.critical(
-            "A personal access token with the note '{}' already exists. "
+            "A repo access token with the note '{}' already exists. "
             "Either delete it or choose another note using the option "
             "'--note'.".format(note))
         sys.exit(1)
+    else:
+        repo_access_response = response.json()
+        LOGGER.info("Success!")
+    return gh_repo["full_name"], auth_response["token"], \
+        repo_access_response["token"]
+
+
+def _setup_travis_ci(gh_repo_name, auth_token, repo_access_token):
+    # travis-ci.org is the open source endpoint only! We will need to hit
+    # travis-ci.com for private projects!
+
+    # Headers for API v2. This is only necessary because generating a Travis
+    # API token from a GitHub Token isn't possible in the API v3 yet. This way
+    # is recommended as per:
+    # https://github.com/travis-ci/travis-ci/issues/9273
+    headers_v2_only = {
+        'User-Agent': 'Memote',
+        'Accept': 'application/vnd.travis-ci.2+json',
+    }
+
+    # Generate Travis API token:
+    try:
+        LOGGER.info("Generating Travis API token.")
+        response = requests.post(
+            "https://api.travis-ci.org/auth/github",
+            headers=headers_v2_only,
+            data={'github_token': auth_token}
+        )
+        response.raise_for_status()
+    except HTTPError:
+        LOGGER.critical(
+            "Something is wrong with the generated APIv3 authentication token "
+            "or you did not link your GitHub account on "
+            "'https://travis-ci.org/'!")
+        sys.exit(1)
+    else:
+        LOGGER.info("Success!")
+        travis_api_token = response.json()["access_token"]
+
+    # Headers for API v3!
+    headers = {
+        'Travis-API-Version': '3',
+        'Authorization': 'token {}'.format(travis_api_token),
+        'User-Agent': 'Memote Query'
+    }
+
+    # Authenticate the User on Travis
     try:
         LOGGER.info("Authorizing with TravisCI.")
-        travis = TravisPy.github_auth(auth.token)
-        t_user = travis.user()
-    except TravisError:
+        response = requests.get(
+            "https://api.travis-ci.org/user", headers=headers
+        )
+        response.raise_for_status()
+    except HTTPError:
         LOGGER.critical(
             "Something is wrong with the generated token or you did not "
             "link your GitHub account on 'https://travis-ci.org/'!")
         sys.exit(1)
-    LOGGER.info("Synchronizing repositories.")
-    while not t_user.sync():
-        sleep(0.1)
+    else:
+        LOGGER.info("Success!")
+        t_user = response.json()
+
+    # Synchronize a User's projects between GitHub and Travis
+    LOGGER.info("Synchronizing user projects between GitHub and Travis.")
+    synced = False
+    while not synced:
```

Comment:
This is a bit odd, why do we need to retry indefinitely?

Reply:
The retry loop was part of the original implementation which I did not want to meddle with.

Should I terminate the loop after X iterations?

Reply:
Ah.. that's hard to answer, so then I think we need to either dig into the original implementation to understand why this was required in the first place, or just leave it as it is.

Reply:
Probably some number of max tries would be better. Maybe a total of 30 or 60 seconds is a good number.

Reply:
Should be fixed in the newest commit.